app/controllers/assignments_controller.rb

```diff
@@ -14,6 +15,10 @@ def index
   def new
     @week = Week.find(params[:week])
 
+    if Assignment.where(week: @week).count >= ASSIGNMENTS_PER_WEEK
```

Comment:
With the constant in there, this is starting to look kind of nasty, eh?

Let's move it into week:

```ruby
def fully_assigned?
  assignments.count >= ASSIGMENTS_PER_WEEK # You don't have to namespace this if the constant is in the same class
end
```
Then you can simplify this line quite a bit:
```ruby
if @week.fully_assigned?
```